crates/interpreter/src/exec.rs

```diff
@@ -728,7 +731,7 @@ enum ThreadResult<'m> {
 
 impl<'m> Thread<'m> {
     fn new(parser: Parser<'m>, frames: Vec<Frame<'m>>) -> Thread<'m> {
```

Comment:
I'm not sure why I made this take a stack of frames, it seems a thread is always created with exactly one frame. We could change the type of this function to take `frame: Frame<'m>` instead of `Vec<Frame<'m>>`. This could be done in another PR against the `main` branch.

Reply:
Done in #612.